src/configuration-requestor.ts

```diff
@@ -18,16 +21,40 @@ export default class ConfigurationRequestor {
 
     await this.flagConfigurationStore.setEntries(configResponse.flags);
     const flagsHaveBandits = Object.keys(configResponse.bandits ?? {}).length > 0;
-    const banditStoreProvided = !!this.banditConfigurationStore;
-    if (flagsHaveBandits && banditStoreProvided) {
+    const banditStoresProvided = Boolean(
+      this.flagBanditConfigurationStore && this.banditModelConfigurationStore,
+    );
+    if (flagsHaveBandits && banditStoresProvided) {
+      // Map bandit flag associations by flag key for quick lookup (instead of bandit key as provided by the UFC)
```

Comment:
I wonder if we can change the server response instead? Indexing by bandit key seems useless for clients and every client now has to do this reshuffling.

Reply:
Yeah, the server response isn't the most useful. It was designed to parallel to `flags` section, where the top-level keys are the keys of the named entity (in this case, bandits).

We already associate the `CONTEXTUAL_BANDIT` algorithm type with variations, so we _could_ deduce which flags have bandits without the `bandits` section at all.

The reason we have this top-level section is to make it really easy to know if we have bandits, which I don't think is that valuable after all if we're traversing the flags anyway. We also could generate a `hasBandits` or `numBandits` field.

Alas, future us!